src/vs/workbench/contrib/chat/common/chatModel.ts

```diff
@@ -522,23 +532,44 @@ export class Response extends AbstractResponse implements IDisposable {
 			this._updateRepr(quiet);
 		} else if (progress.kind === 'textEdit' || progress.kind === 'notebookEdit') {
 			// merge edits for the same file no matter when they come in
-			const groupKind = progress.kind === 'textEdit' ? 'textEditGroup' : 'notebookEditGroup';
+			const notebookUri = CellUri.parse(progress.uri)?.notebook;
+			const uri = notebookUri ?? progress.uri;
+			const groupKind = progress.kind === 'textEdit' && !notebookUri ? 'textEditGroup' : 'notebookEditGroup';
 			let found = false;
 			for (let i = 0; !found && i < this._responseParts.length; i++) {
 				const candidate = this._responseParts[i];
-				if (candidate.kind === groupKind && !candidate.done && isEqual(candidate.uri, progress.uri)) {
-					candidate.edits.push(progress.edits as any);
+				if (candidate.kind === groupKind && !candidate.done && isEqual(candidate.uri, uri)) {
+					if (candidate.kind === 'textEditGroup') {
+						candidate.edits.push(progress.edits as any);
+					} else if (candidate.kind === 'notebookEditGroup') {
+						progress.edits.forEach(edit => {
+							if (TextEdit.isTextEdit(edit)) {
+								candidate.edits.push({ uri: progress.uri, edit });
+							} else {
+								candidate.edits.push(edit);
+							}
+						});
+					}
 					candidate.done = progress.done;
 					found = true;
 				}
 			}
 			if (!found) {
-				this._responseParts.push({
-					kind: groupKind,
-					uri: progress.uri,
-					edits: [progress.edits as any],
-					done: progress.done
-				});
+				if (groupKind === 'textEditGroup') {
+					this._responseParts.push({
+						kind: groupKind,
+						uri,
+						edits: [progress.edits as any],
+						done: progress.done
+					});
+				} else {
+					this._responseParts.push({
+						kind: groupKind,
+						uri,
+						edits: progress.edits.map(edit => TextEdit.isTextEdit(edit) ? { uri: progress.uri, edit } : edit),
```

Comment:
You effectively do twice here and on L545. Could probably pull it up, set `edits = groupKind === 'textEditGroup' ? progress.edits : progress.edits.map(...)` and avoid the if/else here and above

Reply:
yes i could, but i just wanted to make the types clear, else we have to cast to any and thats not great.

Reply:
Will try to clean this up.

Reply:
Fixed